### PR TITLE
lib: move units from machines to lib

### LINKED ISSUE #8572: RFE: cockpit.format_bytes between all units
### Add support for:
- conversion between all units (i.e.  MiB to GiB, TiB to B, etc.)
- converting to and picking the highest unit.

#### Implementation
I already implemented this in machines pkg (for factor 1024): 
https://github.com/cockpit-project/cockpit/blob/master/pkg/machines/helpers.es6#L88

##### Methods:
``` js
convertToUnit(input, inputUnit, outputUnit)
convertToUnitVerbose(input, inputUnit, outputUnit)
convertToBestUnit(input, inputUnit)
```
##### Units
There are units to choose from, although they are not a prerequisite (string can be used instead, e.g. `"KiB"`)
``` js
const units = {}
```

### To Resolve
I am not sure how the API should look like
- should we enhance `cockpit.format_bytes` in http://cockpit-project.org/guide/latest/cockpit-util.html ?
- should we add another function or multiple functions ?

### PR DESCRIPTION
**Pull request by suomiy**
resolves #8572

### CODE REVIEW

File: pkg/machines/components/vmDisksTab.jsx [hunk 1]
```diff
@@ -21,7 +21,8 @@ import React, { PropTypes } from 'react';
 import cockpit from 'cockpit';
 import { Listing, ListingRow } from 'cockpit-components-listing.jsx';
 import { Info } from './inlineNotification.jsx';
-import { convertToUnit, toReadableNumber, units } from "../helpers.es6";
+import { toReadableNumber } from "../helpers.es6";
```
Comment (mareklibra) [inline pkg/machines/components/vmDisksTab.jsx]: just nitpicking: rest of imports uses single quotes `'` instead of `"`.

**Comment by suomiy** [comment]
linter issues fixed and tests passed

**Comment by martinpitt** [comment]
I have similar questions like @mareklibra; let's discuss the API in #8572 first; as-is it appears to be too redundant and inflexible to propagate this as a general API.

**Comment by martinpitt** [comment]
Right, but my point is, it should not get used more in its current form, as it's highly redundant with the existing API. (See #8572)

**Comment by martinpitt** [comment]
This isn't being used outside of machines, and I'm also not happy about this. This has been stuck for a fair while, so let's close this for now.

**Pull request closed without merging**